python/cudf/cudf/core/series.py

```diff
@@ -538,7 +544,13 @@ def __init__(
                 data = {}
 
         if not isinstance(data, ColumnBase):
-            data = column.as_column(data, nan_as_null=nan_as_null, dtype=dtype)
+            data = column.as_column(
+                data,
+                nan_as_null=nan_as_null,
+                dtype=dtype,
+            )
+            if copy:
+                data = data.copy(deep=True)
```

Comment:
Why not just copy the incoming data if copy is asked for, rather than building a column and then copying it? 

Also, can we add coverage here?

Reply:
> Why not just copy the incoming data if copy is asked for, rather than building a column and then copying it?

We don't know if `data` at this point has `.copy` API or `deep` parameter. `copy` also would be really needed if the `data` is on device memory only, which is what was being done previously by just passing through `as_column` and only adhering to `copy` parameter in the case where `data` has a `__cuda_array_interface__`. After Ashwin insisted on not introducing `copy` parameter to `as_column`, I had to move it here.